### ISSUE TITLE
What stops the model learning to predict uniform distributions?

### ISSUE THREAD

**Issue by ocharles** [bluedream1121/self-sca-ori#5]
Hi,

First, let me start by saying congratulations on this work! It's a nice application of self-supervised learning.

I'm curious, what stops your models just learning to predict uniform distributions? Your goal is to align two histograms, but if the model just predicts a flat histogram (equal values for all buckets), then shifting this distribution would be a pathological solution. I ask because this seems to be exactly the problem my own implementation of this work produces. One option is to add an entropy loss to promote non-uniform distributions, but you don't seem to need to do that. Do you have a feeling as to why your work _doesn't_ end up just predicting a uniform pose histogram? I'm primarily interested in orientation.

Thanks!

**Comment by bluedream1121** [comment]
Hello, ocharles. Thank you for your interest!

I also experienced the case of uniform pose distribution case when I work this project.

For the reason of the trivial solution, e.g., flat uniform distribution, we use softmax function during training to facilitate a modality.

Thanks!

**Comment by ocharles** [comment]
Ah, I guess it's the temperature that matters there? (I note in the paper you use 20). Thanks for your response!

**Comment by bluedream1121** [comment]
Yes! the temperature value was important to train the network successfully.

**Comment by ocharles** [comment]
Cool, thanks!
